engine/spverrors/error.go

```diff
@@ -0,0 +1,157 @@
+package spverrors
```

Comment:
With these changes now, shouldn't Errors in models be removed? I dont see them being used anywhere now

Reply:
Maybe, but I wanted to present the key concept for now. 
It you liked it, I would try to adjust the whole codebase to it. 

... Also removing `SPVError`.